### DISCUSSION TITLE
statistics of score_val  of bagged models

### DISCUSSION THREAD

**Discussion by GhsJahanmir** [autogluon/autogluon#4695] [Q&A] [answered]
Dear All,

I have searched extensively in the documentation but was unable to find information on how to obtain the validation (score_val) and test score statistics for bagged models. I am using the medium preset with num_bag_folds=8, and I would like to retrieve and plot the results of different models along with the standard deviation (or other statistics) over the 8 holdout parts of the dataset during the bagging step.

Any ideas or suggestions would be greatly appreciated.

Best Regards,
Ghs

**Comment by LennartPurucker** [comment]
Heyho Ghs!

As far as I know, this is currently not trivially supported. 

You could get some of these values by doing the following (which is not trivial):
* Get the vector of predictions on the validation data (called OOF predictions) via https://auto.gluon.ai/stable/api/autogluon.tabular.TabularPredictor.predict_proba_oof.html
* Split the OOF vector according to the split used by AutoGluon and get your holdout sets. 
	* The hard part of this is determining the original split used to create one OOF vector. You could do this by mirroring the [split made in AutoGluon](https://github.com/autogluon/autogluon/blob/master/core/src/autogluon/core/utils/utils.py) and then using these splits to determine which part of the OOF were holdout parts. 
	* Alternatively, you could save the splits here: https://github.com/autogluon/autogluon/blob/master/core/src/autogluon/core/models/ensemble/bagged_ensemble_model.py#L657

Note that this will not work easily for repeated cross-validation, as we store all OOF predictions in one vector. Computing more statistics (or saving the OOF predictions more verbosely) would require larger code changes and is currently not easily supported.

**Reply by Innixma** [reply, marked as answer]
Note that native support for getting the individual fold predictions is planned for the 1.3 release.
